### DISCUSSION TITLE
Difficulty managing containers

### DISCUSSION THREAD

**Discussion by jpringle1** [i3/i3#4654] [Q&A] [answered]
I can't figure out how to reliably tab/stack windows into the desired containers, even with the user guide.

lets say i have 4 independent windows, like this:
```
+--1--+--2--+--3--+--4--+
|     |     |     |     |
|     |     |     |     |
+-----+-----+-----+-----+
```

Then i decide i want the first two to stack together, and the last two to be split horizontally, so the end result is this:
```
+-----1------+------3------+
+-----2------+             |
|            |------4------|
|            |             |
+------------+-------------+
```

How could i achieve this in the fewest amount of keystrokes?
I can't figure out how to do this at all, never mind doing it easily.

**Comment by orestisfl** [comment, marked as answer]
First of all, you need to understand that i3 stores windows in a tree: https://i3wm.org/docs/userguide.html
Once you start thinking in terms of trees, it becomes easier to arrange windows.

In your case, you first stack all windows, then focus 3, move right, focus 4, move right, switch layout to split v

In tree terms your first layout is H[1,2,3,4] and the second one is H[S[1,2],V[3,4]]

**Reply by jpringle1** [reply]
This is helpful, but I've ran into a problem which may have been the cause of my confusion in the first place:

mod+v and mod+h don't do anything, even if i assign split v/h to different key combos. 
All that happens is a flicker of the screen. I am using the default config file for testing purposes, so i don't think that can be the problem. Any ideas?

FYI, i am running i3 in KDE, in conjunction with some plasma services (taskbar, mainly). I suspect this might be the issue but no idea exactly why.

**Reply by orestisfl** [reply]
> FYI, i am running i3 in KDE, in conjunction with some plasma services (taskbar, mainly). I suspect this might be the issue but no idea exactly why.

In that case, maybe start without KDE and see if the problem still persists

**Reply by jpringle1** [reply]
> > FYI, i am running i3 in KDE, in conjunction with some plasma services (taskbar, mainly). I suspect this might be the issue but no idea exactly why.
> 
> In that case, maybe start without KDE and see if the problem still persists

Hmmm that didn't work either, no idea where to go from here. Any ideas?
